### ISSUE TITLE
SBR can not be deleted if backend service has been deleted

### ISSUE THREAD

**Issue by qibobo** [redhat-developer/service-binding-operator#832]
Steps to reproduce this issue:
1 create service 
```
cat <<EOF | kubectl apply -f -
apiVersion: ibmcloud.ibm.com/v1
kind: Service
metadata:
    name: thetranslator
    annotations:
      ibmcloud.ibm.com/self-healing: enabled
spec:
    plan: standard 
    serviceClass: language-translator
EOF

cat <<EOF | kubectl apply -f -
apiVersion: ibmcloud.ibm.com/v1
kind: Binding
metadata:
    name: thetranslator
spec:
    serviceName: thetranslator
EOF
```
2 create ksvc as the application
```
cat <<EOF | kubectl apply -f -
apiVersion: serving.knative.dev/v1
kind: Service 
metadata:
  name: knative-vcap
spec:
  template:
    metadata:
      annotations:
        autoscaling.knative.dev/minScale: "0"
    spec:
      containers:
        - image: docker.io/qibobo/go4autoscaler:latest
          ports:
          - containerPort: 8080
EOF
```
3 create SBR
```
cat <<EOF | kubectl apply -f -
apiVersion: operators.coreos.com/v1alpha1
kind: ServiceBinding
metadata:
  name: binding-request-knative
spec:
  application:
    group: serving.knative.dev
    version: v1
    resource: services
    name: knative-vcap
  services:
  - group: ibmcloud.ibm.com
    version: v1alpha1
    kind: Binding
    name: thetranslator
    envVarPrefix: serviceprefix
    id: thetranslator
  customEnvVar:
    - name: VCAP_SERVICES
      value: '{"language-translators":[{"credentials":{{json .thetranslator.status.secretName}},"name":"thetranslator","plan":"standard"}]}'
    - name: TRANSLATOR_NAME
      value: "thetranslator"
    - name: URL
      value: '{{.thetranslator.status.secretName.url}}'
EOF
```
4 check the SBR status. 
```
status:
  conditions:
  - lastHeartbeatTime: "2021-01-06T05:49:36Z"
    lastTransitionTime: "2021-01-06T05:49:18Z"
    status: "True"
    type: CollectionReady
  - lastHeartbeatTime: "2021-01-06T05:49:36Z"
    lastTransitionTime: "2021-01-06T05:49:18Z"
    status: "True"
    type: InjectionReady
  - lastHeartbeatTime: "2021-01-06T05:49:36Z"
    lastTransitionTime: "2021-01-06T05:49:18Z"
    status: "True"
    type: Ready
  secret: binding-request-knative
```

5 delete the binding
`kubectl delete binding.ibmcloud thetranslator`
6 check the SBR status again
```
status:
  conditions:
  - lastHeartbeatTime: "2021-01-06T05:51:18Z"
    lastTransitionTime: "2021-01-06T05:51:12Z"
    message: bindings.ibmcloud.ibm.com "thetranslator" not found
    reason: ServiceNotFound
    status: "False"
    type: CollectionReady
  - lastHeartbeatTime: "2021-01-06T05:51:18Z"
    lastTransitionTime: "2021-01-06T05:51:12Z"
    status: "False"
    type: InjectionReady
  - lastHeartbeatTime: "2021-01-06T05:51:18Z"
    lastTransitionTime: "2021-01-06T05:51:12Z"
    status: "False"
    type: Ready
  secret: binding-request-knative
```
7 try to delete SBR but get hang
```
kubectl delete sbr binding-request-knative
servicebinding.operators.coreos.com "binding-request-knative" deleted

```

**Comment by qibobo** [comment]
I will provide a PR for it.

**Comment by pedjak** [comment]
Is this still the issue? https://github.com/redhat-developer/service-binding-operator/pull/855 was submitted but then closed without merging it. Please confirm.

**Comment by qibobo** [comment]
@pedjak 
@cdlliuy has another PR for it.
https://github.com/redhat-developer/service-binding-operator/pull/862

**Comment by wtrocki** [comment]
Was this solved?

**Comment by pedjak** [comment]
I think it is, but can you please verify @wtrocki ?

**Comment by wtrocki** [comment]
Yes. Sounds like duplicate that was solved in 0.7.0
